docs/source/daml/intro/13_Interfaces.rst

```diff
@@ -0,0 +1,252 @@
+party has the power to transfer ownership of the contract to a different party
+(subject to that party agreeing to the transfer!). Daml Interfaces provide a
+way to abstract those behaviors into a Daml type.
+
+.. hint::
+
+  Remember that you can load all the code for this section into a folder called
+  ``intro13`` by running ``daml new intro13 --template daml-intro-13``
+
+
+Context
+-------
+
+First, define some templates:
+
+.. literalinclude:: daml/daml-intro-13/daml/Cash.daml
+  :language: daml
+  :start-after: -- TEMPLATE_CASH_DEF_BEGIN
+  :end-before: -- TEMPLATE_CASH_DEF_END
+
+.. literalinclude:: daml/daml-intro-13/daml/Cash.daml
+  :language: daml
+  :start-after: -- TEMPLATE_CASH_TF_DEF_BEGIN
+  :end-before: -- TEMPLATE_CASH_TF_DEF_END
+
+These declarations from ``intro13/daml/Cash.daml`` define ``Cash`` as a simple
+template with an issuer, an owner, a currency, and an amount. A ``Cash``
+contract grants its ``owner`` the choice ``ProposeCashTransfer``, which allows
+the owner to propose another party, the ``newOwner``, to take over ownership of
+the asset.
+
+This is mediated by the ``CashTransferProposal`` template, which grants two
+choices to the new owner: ``AcceptCashTransferProposal`` and
+``RejectCashTransferProposal``, each of which archives the
+``CashTransferProposal`` and creates a new ``Cash`` contract; in the former case
+the ``owner`` of the new ``Cash`` will be ``newOwner``, in the latter, it will
+be the existing ``owner``. Finally, the existing ``owner`` also has the choice
+``WithdrawCashTransferProposal``, which archives the proposal and creates a new
+``Cash`` contract with identical contents to the original one.
+
+Overall, the effect is that a ``Cash`` contract can be transferred to another
+party, if they agree, in two steps.
+
+The declarations from ``intro13/daml/NFT.daml`` declare the templates ``NFT``
+and ``NFTTransferProposal`` following the same pattern, with names changed where
+appropriate, with the main difference being that an ``NFT`` has a ``url : Text``
+field whereas ``Cash`` has ``currency : Text`` and ``amount : Decimal``.
+
+
+Interface Definition
+--------------------
+
+To abstract this behavior, you will next introduce two interfaces: ``IAsset`` and
+``IAssetTransferProposal``.
+
+.. hint::
+  It is not mandatory to prefix interface names with the letter ``I``, but it
+  can be convenient to tell at a glance whether or not a type is an interface.
+
+.. literalinclude:: daml/daml-intro-13/daml/IAsset.daml
+  :language: daml
+  :start-after: -- INTERFACE_IASSET_DEF_BEGIN
+  :end-before: -- INTERFACE_IASSET_DEF_END
+
+.. literalinclude:: daml/daml-intro-13/daml/IAsset.daml
+  :language: daml
+  :start-after: -- INTERFACE_IASSET_TF_DEF_BEGIN
+  :end-before: -- INTERFACE_IASSET_TF_DEF_END
+
+There are a few things happening here:
+
+1. For each interface, you have defined a ``viewtype``. This is mandatory for all
+   interfaces. All viewtypes must be serializable records. This declaration
```

Comment:
I'd add one sentence to motivate this.

"The viewtype abstracts the read side by providing a uniform way in which implementations of IAsset are represented on the API."

Reply:
That's a great sentence, thank you @bame-da; I'll add it in a new PR. 

> In line with my one comment, I think it would be great to expand this to talk about interfaces on the APIs and in the bindings

I agree that this (or a later chapter) should talk about interfaces over the ledger API and in the bindings, but I think those fall outside the scope of the language/compiler team.